Engine/Main/GameObject.cpp

```diff
@@ -266,7 +318,7 @@ void GameObject::RemoveChild(GameObject* child)
 }
 
 
-ENGINE_API Component* GameObject::CreateComponent(const Component::ComponentType type)
+ENGINE_API Component* GameObject::CreateComponent(const Component::ComponentType type, const ComponentUI::UIType ui_type)
```

Comment:
This function should be overriding the original CreateComponent(), because it should be separated